.github/workflows/build.yml

```diff
@@ -0,0 +1,103 @@
+name: Build
+#
+# This workflow validates that the commit in question can be built successfully
+# in several environments:
+#
+#   Systems:    Ubuntu, MacOS, Windows
+#   Node:       20
+#
+# After the build is successful, the compiled assets are uploaded as an artifact
+# to the workflow run. This allows us to download the compiled assets and use
+# them in other workflows.
+#
+# Note: we need to skip the nx cache b/c it does not contain the compiled assets
```

Comment:
nit: no nx cache on this repo

Reply:
Oh good point! I copied the comment over from CSS.